### PR TITLE
Tighten tests

### PR DESCRIPTION
**Pull request by znicholls**

### CODE REVIEW

**Comment by znicholls** [comment]
Can merge straight into master then you rebase #70, or you can merge this straight into #70, as you wish

File: tests/integration/test_calibrate_mesmer.py [hunk 1]
```diff
@@ -11,6 +11,30 @@
...
 
 
+def _check_dict(first, second, first_name, second_name):
+    for k in first:
+        first_val = first[k]
+        try:
+            second_val = second[k]
+        except KeyError:
+            raise AssertionError(
+                "Key `{}` is in {} but is not in {}".format(k, first_name, second_name)
```
Comment (mathause) [inline tests/integration/test_calibrate_mesmer.py:21]:
```suggestion
                "Key `{}` is in '{}' but is not in '{}'".format(k, first_name, second_name)
```

File: tests/integration/test_calibrate_mesmer.py [hunk 2]
```diff
@@ -63,22 +87,7 @@ def test_calibrate_mesmer(test_data_root_dir, tmpdir, update_expected_files):
...
-                elif isinstance(res_val, np.ndarray):
-                    npt.assert_allclose(res_val, exp_val)
-                elif isinstance(res_val, xr.DataArray):
-                    xrt.assert_allclose(res_val, exp_val)
-                else:
-                    assert res_val == exp_val, k
-
-        _check_dict(res, exp)
+        # check all keys of res match exp
+        _check_dict(res, exp, "res", "exp")
```
Comment (mathause) [inline tests/integration/test_calibrate_mesmer.py:91]:
```suggestion
        _check_dict(res, exp, "result", "expected")
```

File: tests/integration/test_calibrate_mesmer.py [hunk 3]
```diff
@@ -63,22 +87,7 @@ def test_calibrate_mesmer(test_data_root_dir, tmpdir, update_expected_files):
...
+        # check all keys of exp match res
+        _check_dict(exp, res, "exp", "res")
```
Comment (mathause) [inline tests/integration/test_calibrate_mesmer.py:93]:
```suggestion
        _check_dict(exp, res, "expected", "result")
```

**Review by mathause — APPROVED** [review]
:+1:

**Review by leabeusch — APPROVED** [review]
Great, thanks @znicholls! (& @mathause for the review comments)
Keeping this separated from & before #70 seems cleaner to me. I guess this means I'll start collecting first rebasing experiences.

**Pull request merged**